CMakeLists.txt

```diff
@@ -386,6 +383,14 @@ ENABLE_TESTING()
 # It would be great to use GTEST_OUTPUT directly, but I couldn't get it to work.
 set(GOOGLE_TEST_OUTPUT --gtest_output=xml:${CMAKE_BINARY_DIR}/results/)
 
+# amalgamate jsoncpp source at compile time
+# note: jsoncpp has a CMake support that they intent to deprecate and is hard to
+# integrate with.
+execute_process(COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/scripts/amalgamate-jsoncpp.sh ${CMAKE_CURRENT_SOURCE_DIR}/third_party/jsoncpp ${CMAKE_CURRENT_BINARY_DIR}/jsoncpp)
```

Comment:
Why do we have to do this amalgamation?

How else do they build if they are deprecating their CMake support?

Reply:
Mainly meson https://github.com/open-source-parsers/jsoncpp/wiki/Building.

I've made another attempt by using CMake the "proper" way and mostly just `add_subdirectory()` but it was quite hard to control the compilation flags and I think I did not manage to remove the compilation of some internal test executable.

Just using the amalgamated source, like we were doing before, gives more flexibility.

Note that the amalgamation process is done at configure time and that cmake would have to be re-run after updating the submodule.

Reply:
Yeah, this seems fine, it's just a bit weird.